x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_definition/rum_settings.ts

```diff
@@ -4,10 +4,31 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
 import { i18n } from '@kbn/i18n';
 import { PackagePolicyVars, SettingsRow } from '../typings';
 import { isSettingsFormValid, OPTIONAL_LABEL } from '../settings_form/utils';
 
+const arrayRegex = new RegExp(/\[[^\]]*\]/);
+function getAllowedOriginsRt() {
+  return new t.Type<string, string, unknown>(
+    'allowedOriginsRt',
+    t.string.is,
+    (input, context) => {
+      return either.chain(
+        t.string.validate(input, context),
+        (inputAsString) => {
+          return arrayRegex.test(inputAsString)
+            ? t.failure(input, context, 'Invalid value. Array is not allowed')
```

Comment:
Should the failure message use `i18n.translate`?

Reply:
https://github.com/elastic/kibana/pull/137228/commits/5efaea4f8e4deaf1e334b7d00d167decdcaa98ba